src/docs.js

```diff
@@ -6,38 +6,48 @@ let path = require('path');
 let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
-
   options = _.defaults({}, options, {
+    metadata: null,
     schemas: [],
     docsFolder: null,
+    references: [],
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
-  let schemas = options.schemas;
   let tarball = tar.pack();
+  let metadata = options.metadata;
+
+  if (metadata) {
```

Comment:
> So if I understand correctly, each project will have a different menuIndex value. 

Yes... Since each project is responsible for defining it's own menu index, and projects are managed by different people... We can't prevent a menuIndex collision... but in such cases we can resolved it by sorting alphabetically or something... And when we see unwanted menu index we can file a bug for the component and fix it. The whole menu ordering will be sort of a best-effort kind of thing, I don't see any way we can build a menu from distributed components without some sort of aspect like this...

> We can test that default values are not overriden and if it is, then throw an error. We will then have a better idea when we run into issues.

What default values? I might miss the context here...